src/librustc_typeck/astconv.rs

```diff
@@ -609,7 +617,39 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                                 // the arguments don't match up with the parameters, we won't issue
                                 // an additional error, as the user already knows what's wrong.
                                 if arg_count_correct {
-                                    Self::generic_arg_mismatch_err(tcx.sess, arg, kind.descr());
+                                    // We're going to iterate over the parameters to sort them out, and
+                                    // show that order to the user as a possible order for the parameters
+                                    let mut lifetimes = vec![];
+                                    let mut types = vec![];
+                                    let mut consts = vec![];
+
+                                    let params = defs.params.clone();
+                                    for param in params {
+                                        match param.kind {
+                                            GenericParamDefKind::Lifetime => lifetimes.push(param),
+                                            GenericParamDefKind::Type { .. } => types.push(param),
+                                            GenericParamDefKind::Const => consts.push(param),
+                                        }
+                                    }
+
+                                    let mut ordered_params = vec![];
+                                    ordered_params.extend(lifetimes);
+                                    ordered_params.extend(types);
+                                    ordered_params.extend(consts);
+
+                                    Self::generic_arg_mismatch_err(
+                                        tcx.sess,
+                                        arg,
+                                        kind.descr(),
+                                        Some(&format!(
+                                            "Wrong parameter order found. A possible order is: {}",
```

Comment:
```suggestion
                                            "wrong parameter order found. A possible order is: {}",
```
We do not capitalize the first letter of `help` messages, see https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f820a1bcddd804963a03c9f0c79a0071 for an example